src/index.js

```diff
@@ -0,0 +1,220 @@
+    return new Promise((resolve, reject) => {
+      const output = path.join(
+        this.compiledCircuitsPath,
+        `${path.basename(filepath, Extension.Circom)}${Extension.Json}`
+      );
+      exec(`${circomBinary} ${filepath} -o ${output}`, error => {
+        if (error) {
+          return reject(error);
+        }
+
+        resolve();
+      });
+    });
+  }
+
+  async generateProofs() {
+    this.embark.logger.info('Generating proofs...');
+
+    await Promise.all(
+      (await fs.readdir(this.compiledCircuitsPath))
+        .filter(filename => path.extname(filename) === Extension.Json)
+        .map(filename => this.generateProof(filename))
+    );
+  }
+
+  async generateProof(filename) {
+    const filepath = path.join(this.compiledCircuitsPath, filename);
+    const basename = path.basename(filename, Extension.Json);
+
+    const circuit = await this.getCircuit(filepath);
+    const setup = await this.generateSetup(circuit, basename);
+
+    const input = this.circuitsConfig.inputs[basename];
+    if (!input) {
+      return;
+    }
+
+    const witness = circuit.calculateWitness(input);
+    const { proof, publicSignals } = zkSnark.original.genProof(
+      setup.vk_proof,
+      witness
+    );
+    if (zkSnark.original.isValid(setup.vk_verifier, proof, publicSignals)) {
+      return this.generateVerifier(basename);
+    }
+
+    throw new Error(
+      `The proof is not valid for ${basename} with inputs: ${JSON.stringify(
+        input
+      )}`
+    );
+  }
+
+  async getCircuit(filepath) {
+    const definition = JSON.parse(await fs.readFile(filepath, 'utf8'));
+    return new zkSnark.Circuit(definition);
+  }
+
+  async generateSetup(circuit, basename) {
+    const setup = zkSnark.original.setup(circuit);
+    await Promise.all([
+      fs.writeFile(
+        this.proofFilepath(basename),
+        JSON.stringify(setup.vk_proof),
+        'utf8'
+      ),
+      fs.writeFile(
+        this.verifierFilepath(basename),
+        JSON.stringify(setup.vk_verifier),
+        'utf8'
+      )
+    ]);
+
+    return setup;
+  }
+
+  generateVerifier(basename) {
+    return new Promise((resolve, reject) => {
+      const source = this.verifierFilepath(basename);
+      const output = this.verifierContractPath(basename);
+      exec(
+        `${snarkjsBinary} generateverifier --vk ${source} -v ${output}`,
+        error => {
+          if (error) {
+            return reject(error);
+          }
+
+          resolve();
+        }
+      );
+    });
+  }
+
+  async addVerifiersToContracts() {
+    (await fs.readdir(this.compiledCircuitsPath))
+      .filter(filename => path.extname(filename) === Extension.Solidity)
+      .map(filename => this.addVerifierToContracts(filename));
+  }
+
+  addVerifierToContracts(filename) {
+    // won't work as desired with "dev embark" (built in monorepo) until after
+    // first prerelease of embark v5
+    if (semver(this.embark.version).major >= 5 /* || true */) {
```

Comment:
I'd create a `isEmbark5` function or something similar, since you repeat the condition
```suggestion
    if (semver(this.embark.version).major >= 5) {
```

Reply:
Done.